### DISCUSSION TITLE
Martensite Variants in Rodrigues vector

### DISCUSSION THREAD

**Discussion by danishTUE** [mtex-toolbox/mtex#2240] [Ask Anything]
Hello,

I am trying to learn MTex for generating martensite variants. As a simple example, I use parent austenite euler angles to be [0,0,0]. When I use MTex to generate martensite variants for this parent orientation and then convert it to Rodrigues vector, I get the variants as infinity. I checked each variant one by one and saw that one of the variants ( I think variant 21) gives the Rodrigues vector as infinity. I am not sure if I am doing something wrong, or is it really the case. Below is my code:

```matlab
cs_fcc = crystalSymmetry('432', [3.6599 3.6599 3.6599], 'mineral', 'Iron fcc');
cs_bcc = crystalSymmetry('432', [2.866 2.866 2.866], 'mineral', 'Iron bcc');
my_ori = orientation.byEuler(0,0,0,cs_fcc);
p2c = orientation.KurdjumovSachs(cs_fcc,cs_bcc);
my_mart_ori = variants(p2c, my_ori);
my_mart_rods = my_mart_ori.Rodrigues;
my_rods = my_mart_rods.xyz
``` 
Is what I am doing correct? If yes, would the Rodriguez vector convention not work for this case? How can I express the orientations of these variants in Rodriguez vector convention?

Many thanks in advance.

Regards,

Danish

**Comment by nyyssont** [comment]
Hello, you might want to try reversing the command `my_mart_ori = variants(p2c, my_ori);` to `my_mart_ori = variants(my_ori, p2c);`

I know, I told you to go by the documentation... in this case there seems to have been a change at some point on how the command is supposed to be given.

**Reply by danishTUE** [reply]
Hello Tuomo,

Thanks for your response. 

As I see from the documentation: [https://mtex-toolbox.github.io/orientation.variants.html](url), the orientational relationship (i.e `p2c` in this case) is the first argument of the function `variants`. If I understood your comment correctly, its not correct in this documentation and the correct way to use this function is by giving the orientational relationship as the second argument to `variants` (as you have pointed out). Is my understanding correct?  

Regards,

Danish

**Reply by nyyssont** [reply]
Yes, your understanding is correct.

**Reply by nyyssont** [reply]
Well... looks like I'm mistaken again. `my_mart_ori = variants(p2c, my_ori);` seems to produce correct orientation. You have the right syntax from the start: it is some other problem with producing Rodrigues vectors.

**Reply by danishTUE** [reply]
Thanks Tuomo for the clarification. Yes, I am not sure why Rodrigues vectors of one of the variants is infinity. I used DREAM-3D plugin to convert the Euler Angles of the same variant to Rodrigues Vector and I get infinity there as well. Can this probably be the limitation of the Rodrigues convention?

**Reply by ralfHielscher** [reply]
This is a problem with the Rodrigues convention. All Rotations with rotational angle 180 degree are assigned to infinity. In the case of cubic symmetry you can overcome this issue by projecting the orientations the fundamental zone before converting into Rodrigues vectors, i.e., 
```
my_mart_ori = project2FundamentalRegion(variants(p2c, my_ori));
my_mart_rods = my_mart_ori.Rodrigues;
my_rods = my_mart_rods.xyz
```

**Reply by danishTUE** [reply]
Dear Dr. Ralf,

Many thanks for your reply and the explanation. I tried generating the variants by projecting them to the fundamental zone (new way) and compared them with how I generated them previously (old way). I see that the Euler angles (and thus the Rodrigues vectors) of the variants are different in both cases. To understand better, I tried plotting these old and new variants in 3D with crystal shapes. Here is my code:
```matlab
cs_fcc = crystalSymmetry('432', [3.6599 3.6599 3.6599], 'mineral', 'Iron fcc');
cs_bcc = crystalSymmetry('432', [2.866 2.866 2.866], 'mineral', 'Iron bcc');
my_ori = orientation.byEuler(0,0,0,cs_fcc);
p2c = orientation.KurdjumovSachs(cs_fcc,cs_bcc);
% old way of generating variants
my_mart_ori_old = variants(p2c, my_ori);
% new way of generating variants
my_mart_ori_new = project2FundamentalRegion(variants(p2c, my_ori));
my_mart_rods_old = my_mart_ori_old.Rodrigues;
my_mart_rods_new = my_mart_ori_new.Rodrigues;
my_rods_old = my_mart_rods_old.xyz;
my_rods_new = my_mart_rods_new.xyz;
cS_old = crystalShape.cube(cs_bcc);
cS_new = crystalShape.cube(cs_bcc);
x=0;y=0;
for j=1:24
    % Plotting old variants
    shift_old = vector3d(x,y,0);
    pcS=plot(shift_old+my_mart_ori_old(1,j) * cS_old,'FaceColor','white');
    alpha(pcS,0.5);
    hold on
    % Plotting reference planes on old variants
    p_100 = cS_old(Miller(1,0,0,cS_old.CS));
    p_010 = cS_old(Miller(0,1,0,cS_old.CS));
    plot(shift_old+my_mart_ori_old(1,j) * p_100,'FaceColor','r');
    plot(shift_old+my_mart_ori_old(1,j) * p_010,'FaceColor','yellow');
    % Plotting new variants
    shift_new = shift_old+vector3d(0,0,1.5);
    pcS=plot(shift_new+my_mart_ori_new(1,j) * cS_new,'FaceColor','white');
    alpha(pcS,0.2);
    % Plotting reference planes on new variants
    p_100 = cS_new(Miller(1,0,0,cS_new.CS));
    p_010 = cS_new(Miller(0,1,0,cS_new.CS));
    plot(shift_new+my_mart_ori_new(1,j) * p_100,'FaceColor','r');
    plot(shift_new+my_mart_ori_new(1,j) * p_010,'FaceColor','yellow');
    x=x+1.5;
end 
hold off
```  

I observe the following: the orientations of the variants in space w.r.t the sample in both the methods are symmetrically equivalent but they are **not exactly same**. (The reference planes of (100) and (010) of the variants in two sets do not match in the generated image). Since I intend to use the Rodrigues vector of the martensite variants as input for my crystal plasticity model for deformation, I presume that using the variants generated in the new way will not exactly be the same as using variants generated in the old way. Is my understanding correct? (Kindly excuse my lack of knowledge. I am not well-versed in crystallography. I briefly read about projecting to fundamental regions. Superficially, I understood that it kind of reduces the orientation space by replacing the old orientation with a symmetrically equivalent orientation which lies in the fundamental region)

**Reply by ralfHielscher** [reply]
I would put it that way: symmetrically equivalent Rodrigues vectors use different numbers to describe the same physical property. The consequence should be. Any crystal plasticity code should give the same result not matter

- which of the different numbers are used as input
- whether orientations are parameterized as Rodrigues vectors, quaternion, Euler angles, ...

as the same physical input should result in the same output. Maybe the usage of Rodrigues vectors as parametrisation in your crystal plasticity code was not a good choice?

Ralf.

**Reply by danishTUE** [reply]
Dr. Ralf, 

Many thanks for your explanation. Ofcourse, it seems to me now that Rodrigues vector may not be a good choice because of these limitations. I just have one follow-up question and then I am done. It would be helpful if you can kindly answer it. It is as follows:

I understand that symmetrically equivalent Rodrigues vectors would lead to the same physical quantity. However, if the physical quantity is a tensor (or a vector), then would the components of the physical quantity in the crystal coordinate system be the same in two the cases. For e.g. Under the same imposed global strain, would the slip (or projected strain) on a particular (111) slip plane be the same in both cases (although I do understand that they result in the same strain tensor globally)?

Regards,

Danish

**Reply by ralfHielscher** [reply]
Tensors that describe physical properties of single crystals must have the same symmetry properties as the crystal, i.e., the tensor needs to be invariant with respect to all symmetry operations. This is well understood for e.g. elasticity tensors. 
Ralf.
